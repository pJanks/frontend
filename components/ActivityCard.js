import React from "react";
import { View, Text, StyleSheet } from "react-native";

export default function ActivityCard({ date }) {
  console.log(date);
  return (
    <View>
      <Text>{date}</Text>
      <Text>hi</Text>
    </View>
  );
}

// const styles = StyleSheet.create({
//   picker: {
//     height: 40,
//     width: 250,
//     borderColor: 'gray',
//     borderWidth: 1,
//     backgroundColor: '#fff',
//   }
// });
