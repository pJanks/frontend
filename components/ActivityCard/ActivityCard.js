import React from "react";
import { View, Text, StyleSheet, Button } from "react-native";

export default function ActivityCard({ date, location, activity, forecast, navigation }) {
  return (
    <View style={ styles.activityCard }>
      <Text style={ styles.header }>Scheduled Activity:</Text>
      <Text style={ styles.allActivityCardText }>
        <Text style={ styles.label } >Scheduled on: </Text>
        <Text>{ `${date}\n`}</Text>
        <Text style={ styles.label } >{ activity } at/in: </Text>
        <Text>{ `${location}\n`}</Text>
        <Text style={ styles.label } >Forecast: </Text>
        <Text>{ `${forecast}\n\n` }</Text>
      </Text>
      <Button
        title='View Activity'
        onPress={() => navigation.navigate('ActivityDetails')}
        color ='#000'>
      </Button>
    </View>
  );
}

const styles = StyleSheet.create({
  activityCard: {
    borderWidth: 2,
    borderColor: "#000",
    borderRadius: 25,
    height: 260,
    margin: 30,
    padding: 20,
    width: '86%',

  },

  allActivityCardText: {
    fontSize: 20
  },

  bottomOfCard: {
    marginBottom: 5000
  },

  header: {
    fontSize: 25,
    marginBottom: 10,
    paddingHorizontal: '7%',
    textDecorationLine: 'underline'
  },

  label: {
    fontWeight: 'bold'
  }
});
