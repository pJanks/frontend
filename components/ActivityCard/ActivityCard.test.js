import React from 'react';
import ActivityCard from './ActivityCard';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const mockData = {
      "id": 1,
      "date": "2020-04-06",
      "location": "Denver, CO",
      "forecast": "Overcast",
      "forecast_img": "cloudy",
      "activity": {
          "id": 1,
          "name": "Hiking",
          "muscle_groups": [
              {
                  "id": 1,
                  "name": "Biceps",
                  "exercise_set": [
                      {
                          "id": 1,
                          "name": "Pull Ups",
                          "description": "Pull yourself up ya dingus",
                          "img_url": "fake_img"
                      }
                  ]
              },
              {
                  "id": 2,
                  "name": "Forearms",
                  "exercise_set": [
                      {
                          "id": 3,
                          "name": "Push Ups",
                          "description": "Push yo' self",
                          "img_url": "some_img"
                      },
                      {
                          "id": 4,
                          "name": "Small Arm Circles",
                          "description": "Feel the burn",
                          "img_url": "some_img"
                      },
                  ]
              }
          ]
      },
      "user": {
          "id": 1,
          "first_name": "user",
          "last_name": "name"
      },
      "created_at": "2020-04-06T03:23:26.122544Z",
      "updated_at": "2020-04-06T03:23:26.122555Z"
  };

  const activityCard = renderer.create(<ActivityCard
  date={ mockData.date }
  location={ mockData.location }
  activity={ mockData.activity.name }
  forecast={ mockData.forecast }
  activityId={ mockData.id }
  />).toJSON();
  
  expect(activityCard).toMatchSnapshot();
});
