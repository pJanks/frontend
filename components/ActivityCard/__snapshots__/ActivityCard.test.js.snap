// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderColor": "#07407b",
      "borderRadius": 25,
      "borderWidth": 2,
      "margin": 10,
      "padding": 20,
      "width": "85%",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 25,
        "marginBottom": 10,
        "textAlign": "center",
        "textDecorationLine": "underline",
      }
    }
  >
    Hiking
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "margin": 2,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Scheduled on: 
    </Text>
    <Text
      style={
        Object {
          "fontSize": 16,
        }
      }
    >
      2020-04-06
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "margin": 2,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Hiking
       at/in: 
    </Text>
    <Text
      style={
        Object {
          "fontSize": 16,
        }
      }
    >
      Denver, CO
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "margin": 2,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Forecast: 
    </Text>
    <Text
      style={
        Object {
          "fontSize": 16,
        }
      }
    >
      Overcast
    </Text>
  </View>
  <View
    style={
      Object {
        "marginTop": 15,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#07407b",
              },
            ]
          }
        >
          View Activity
        </Text>
      </View>
    </View>
  </View>
</View>
`;
